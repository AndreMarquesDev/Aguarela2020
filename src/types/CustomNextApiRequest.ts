import type { NextApiRequest } from 'next';
import type { FormPostRequestBody } from './FormPostRequestBody';

export type CustomNextApiRequest = {
    body: FormPostRequestBody;
} & NextApiRequest;
