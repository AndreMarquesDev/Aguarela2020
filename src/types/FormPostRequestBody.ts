import type { FormValues } from './FormValues';

export type FormPostRequestBody = {
    isPlaywright: boolean;
} & FormValues;
