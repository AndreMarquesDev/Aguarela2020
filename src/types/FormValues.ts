export type FormValues = {
    name: string;
    email: string;
    message: string;
    brand: string;
    subject: string;
};
