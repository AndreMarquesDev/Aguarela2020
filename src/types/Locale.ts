export enum Locale {
    Pt = 'pt',
    En = 'en',
}
