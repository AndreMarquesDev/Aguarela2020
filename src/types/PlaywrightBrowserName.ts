export enum PlaywrightBrowserName {
    Chromium = 'chromium',
    Firefox = 'firefox',
    Webkit = 'webkit',
}
