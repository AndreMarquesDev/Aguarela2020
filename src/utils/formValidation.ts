export enum FieldTypes {
    Name = 'name',
    Email = 'email',
    Textarea = 'message',
    Brand = 'brand',
    Subject = 'subject',
}
export type FormState = {
    name: string;
    email: string;
    message: string;
    brand: string;
    subject: string;
};

export const validateName = (value: string): boolean => {
    const nameRegex =
        // eslint-disable-next-line regexp/no-unused-capturing-group, regexp/prefer-plus-quantifier, regexp/prefer-character-class, regexp/no-super-linear-backtracking, regexp/no-useless-escape
        /s*([a-zA-ZáâàãéêèíîìóôòõúûùñçäöüßÁÂÀÃÉÊÈÍÎÌÓÔÒÕÚÛÙÑÇÄÖÜẞ]{1,}([\.,] |[-']| ))+[a-zA-ZáâàãéêèíîìóôòõúûùñçäöüßÁÂÀÃÉÊÈÍÎÌÓÔÒÕÚÛÙÑÇÄÖÜẞ]+\.?\s*$/;

    return nameRegex.test(value);
};

export const validateEmail = (value: string): boolean => {
    const emailRegex =
        // eslint-disable-next-line regexp/no-useless-escape, regexp/no-unused-capturing-group, regexp/prefer-d, regexp/use-ignore-case
        /(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;

    return emailRegex.test(value);
};
