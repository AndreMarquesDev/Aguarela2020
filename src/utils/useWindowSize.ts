import { useEffect, useState } from 'react';

type WindowSize = {
    width: number;
    height: number;
};

export const useWindowSize = (): WindowSize => {
    const [windowSize, setWindowSize] = useState<WindowSize>({
        width: 0,
        height: 0,
    });

    useEffect(() => {
        const handleResize = (): void => {
            // eslint-disable-next-line react-hooks-extra/no-direct-set-state-in-use-effect
            setWindowSize({
                width: window.innerWidth,
                height: window.innerHeight,
            });
        };

        window.addEventListener('resize', handleResize);

        handleResize();

        return (): void => window.removeEventListener('resize', handleResize);
    }, []);

    return windowSize;
};

export enum Breakpoint {
    Desktop = 1280,
    Phablet = 1023,
    Tablet = 767,
    Mobile = 511,
}
