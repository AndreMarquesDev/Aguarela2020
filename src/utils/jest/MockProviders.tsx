/* eslint-disable react/jsx-no-constructed-context-values */
import React, { ReactNode } from 'react';
import { NavLinksContext } from '../../components/context/NavLinksContext';
import { TextsContext } from '../../components/context/TextsContext';
import { Locale } from '../../types/Locale';
import { textsEn, textsPt } from '../texts';

interface MockProvidersProps {
    language?: Locale;
    isMenuOpen?: boolean;
    toggleMenu?: jest.Mock;
    setNavHeight?: jest.Mock;
    children: ReactNode;
}

export const MockProviders = ({
    language = Locale.Pt,
    isMenuOpen = false,
    toggleMenu = jest.fn(),
    setNavHeight = undefined,
    children,
}: MockProvidersProps): JSX.Element => {
    return (
        <TextsContext.Provider
            value={{
                texts: language === Locale.Pt ? textsPt : textsEn,
            }}
        >
            <NavLinksContext.Provider value={{ isMenuOpen, toggleMenu, setNavHeight }}>
                {children}
            </NavLinksContext.Provider>
        </TextsContext.Provider>
    );
};
