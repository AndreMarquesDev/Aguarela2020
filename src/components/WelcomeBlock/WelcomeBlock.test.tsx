import type { RenderResult } from '@testing-library/react';
import { render, screen } from '@testing-library/react';
import * as nextRouter from 'next/router';
import React from 'react';
import { Locale } from '../../types/Locale';
import { MockProviders } from '../../utils/jest/MockProviders';
import { setJestWindowWidth } from '../../utils/jest/setJestWindowWidth';
import { textsEn, textsPt } from '../../utils/texts';
import { Breakpoint } from '../../utils/useWindowSize';
import { WelcomeBlock } from './WelcomeBlock';
import '@testing-library/jest-dom';

// @ts-expect-error ignore for now
nextRouter.useRouter = jest.fn(() => ({
    query: {
        language: 'pt',
    },
}));

const renderComponent = (language: Locale = Locale.Pt): RenderResult => {
    return render(
        <MockProviders language={language}>
            <WelcomeBlock />
        </MockProviders>,
    );
};

describe('<WelcomeBlock />', () => {
    test('renders properly', () => {
        const { container } = renderComponent();

        const title1 = screen.getByText(textsPt.welcome1);
        const title2 = screen.getByText(textsPt.welcome2);
        const title3 = screen.getByText(textsPt.welcome3);
        const textBlock1 = screen.getByText(textsPt.managingsocialMedia);
        const textBlock2 = screen.getByText(textsPt.monitorAndOptimizeProcessAndStrategy);
        const textBlock3 = screen.getByText(textsPt.throughStrategicPlanning);

        expect(title1).toBeInTheDocument();
        expect(title2).toBeInTheDocument();
        expect(title3).toBeInTheDocument();
        expect(textBlock1).toBeInTheDocument();
        expect(textBlock2).toBeInTheDocument();
        expect(textBlock3).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });

    test('renders properly in English', () => {
        const { container } = renderComponent(Locale.En);

        const title1 = screen.getByText(textsEn.welcome1);
        const title2 = screen.getByText(textsEn.welcome2);
        const title3 = screen.getByText(textsEn.welcome3);
        const textBlock1 = screen.getByText(textsEn.managingsocialMedia);
        const textBlock2 = screen.getByText(textsEn.monitorAndOptimizeProcessAndStrategy);
        const textBlock3 = screen.getByText(textsEn.throughStrategicPlanning);

        expect(title1).toBeInTheDocument();
        expect(title2).toBeInTheDocument();
        expect(title3).toBeInTheDocument();
        expect(textBlock1).toBeInTheDocument();
        expect(textBlock2).toBeInTheDocument();
        expect(textBlock3).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });

    test('renders properly on mobile', () => {
        setJestWindowWidth(Breakpoint.Mobile);

        renderComponent();

        const titleMobile = screen.getByText(
            `${textsPt.welcome1}-${textsPt.welcome2}${textsPt.welcome3}`,
        );

        expect(titleMobile).toBeInTheDocument();
    });

    test('renders properly on mobile in English', () => {
        (nextRouter.useRouter as jest.Mock).mockImplementationOnce(() => ({
            query: {
                language: 'en',
            },
        }));

        setJestWindowWidth(Breakpoint.Mobile);

        renderComponent(Locale.En);

        const titleMobile = screen.getByText(
            `${textsEn.welcome1}${textsEn.welcome2}${textsEn.welcome3}`,
        );

        expect(titleMobile).toBeInTheDocument();
    });
});
