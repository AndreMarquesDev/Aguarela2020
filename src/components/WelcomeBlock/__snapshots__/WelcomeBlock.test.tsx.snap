// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomeBlock /> renders properly 1`] = `
<div>
  <section
    class="jsx-1059500243 container"
    data-testid="welcome_section"
  >
    <div
      class="jsx-1059500243 wrapper genericMargins"
    >
      <div
        class="jsx-1059500243 titleBlock"
      >
        <strong
          class="jsx-1059500243 fontXXL"
        >
          Bem
        </strong>
        <strong
          class="jsx-1059500243 fontXXL"
        >
          Vin
        </strong>
        <strong
          class="jsx-1059500243 fontXXL"
        >
          Dos
        </strong>
      </div>
      <div
        class="jsx-1059500243 textBlock"
      >
        <p
          class="jsx-1059500243 bodyText bodyText--white"
          data-testid="textShadowBlock_textBlock1"
        >
          Juntámos a vontade de comunicar ao gosto por fotografar. A Aguarela Digital é uma agência de comunicação que tem como objetivo posicionar e fazer brilhar negócios locais e pequenas empresas no digital, desenvolvendo estratégias de comunicação adequadas às suas necessidades e objetivos.
        </p>
        <p
          class="jsx-1059500243 bodyText bodyText--white"
        >
          Através do planeamento estratégico, criação de uma identidade visual, conteúdo de qualidade e implementação de campanhas digitais, a Aguarela dedica-se a todas as marcas como se fossem suas.
        </p>
        <p
          class="jsx-1059500243 bodyText bodyText--white"
        >
          Do o início ao fim, fazemos o acompanhamento, monitorização e otimização de todo o processo e estratégia de gestão de redes sociais, campanhas paid social (Facebook e Instagram) e paid search (Google Ads), otimização de SEO, criação de conteúdo e consultoria.
        </p>
      </div>
    </div>
  </section>
</div>
`;

exports[`<WelcomeBlock /> renders properly in English 1`] = `
<div>
  <section
    class="jsx-1059500243 container"
    data-testid="welcome_section"
  >
    <div
      class="jsx-1059500243 wrapper genericMargins"
    >
      <div
        class="jsx-1059500243 titleBlock"
      >
        <strong
          class="jsx-1059500243 fontXXL"
        >
          Wel
        </strong>
        <strong
          class="jsx-1059500243 fontXXL"
        >
          Co
        </strong>
        <strong
          class="jsx-1059500243 fontXXL"
        >
          Me
        </strong>
      </div>
      <div
        class="jsx-1059500243 textBlock"
      >
        <p
          class="jsx-1059500243 bodyText bodyText--white"
          data-testid="textShadowBlock_textBlock1"
        >
          We meshed the desire to communicate with the passion for photography. Aguarela Digital is a communication agency that aims to position and make local and small businesses shine in the digital world, developing communication strategies suited to their needs and goals.
        </p>
        <p
          class="jsx-1059500243 bodyText bodyText--white"
        >
          From start to finish we follow up, monitor and optimize the entire process and strategy of social media management, paid social campaigns (Facebook and Instagram) and paid search (Google Ads), SEO optimization, content creation and consulting.
        </p>
        <p
          class="jsx-1059500243 bodyText bodyText--white"
        >
          Through strategic planning, creation of a visual identity, quality content and implementation of digital campaigns, Aguarela dedicates itself to all brands as if they were their own.
        </p>
      </div>
    </div>
  </section>
</div>
`;
